### ISSUE TITLE
cgnal.core.config.path_constructor breaks when parsed path does not contain env variable

### ISSUE THREAD

**Issue by NicolaDonelli** [CGnal/cgnal-core#32]
cgnal.core.config.path_constructor breaks when parsed path does not contain env variable expressed between curly brackets.

This is the current imlementation.
```
path_matcher = re.compile(r"\$\{([^}^{]+)\}")


def path_constructor(
    loader: Union[Loader, FullLoader, UnsafeLoader], node: Node
) -> PathLike:
    """Extract the matched value, expand env variable, and replace the match."""
    value = node.value
    match = path_matcher.match(value)

    if match is None:
        raise SyntaxError("Can't match pattern")

    env_var = match.group()[2:-1]
    return os.environ.get(env_var) + value[match.end():]
```

I'd suggest fixing the return line with:
```
   return os.environ.get(env_var, "") + value[match.end():]
```
but I don't know where exactly this is used and what were the reasons of the former implementation committed by @deusebio on cgnal-data-ingestion in revision 47141b85e8e46cda1189b3009a970cbb1b9a60a2, 08/10/2019, "[BDL-148] Authentication (user/pwd) Configs and Bash Environ Variable Integration", so this might be wrong.
@deusebio

**Comment by deusebio** [comment]
Yes, this is usually caused by the fact that the enviroment variable is not defined. However, the way it breaks makes this not very obvious and rather obscure (why is the missing env variable?)

I believe we should introduce more informative messages here, that - at the very least - signals which env variable is missing

I would not put some defaults if the variable is missing. The type and the value of the defaults may vary from use-case to use-case, and having a default to empty string is rather specific. 

I will address this issue by simply putting some checks. If your problem was not related to missing env variables, please try to come up with a minimal reproducible example that I can look into and post it as a different issue.
